### PR TITLE
Add Team Capes plugin

### PR DESCRIPTION
**Pull request by devinfrench**
Disabled by default. Used by clans when pking.

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesOverlay.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+	@Override
+	public Dimension render(Graphics2D graphics, Point parent)
+	{
+		if (!config.enabled() || client == null || client.getGameState() != GameState.LOGGED_IN)
+		{
+			return null;
+		}
+
+		Map<Integer, Integer> teams = new HashMap<>();
+		client.getPlayers().forEach(
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesOverlay.java]: This seems rather expensive to do once per frame. Maybe schedule it every couple of seconds?

File: runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesOverlay.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+		setPosition(OverlayPosition.TOP_LEFT);
+		setPriority(OverlayPriority.MED);
+		this.client = client;
+		this.config = config;
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics, Point parent)
+	{
+		if (!config.enabled() || client == null || client.getGameState() != GameState.LOGGED_IN)
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesOverlay.java]: Is it only useful to do this if the local player is on a team?

File: runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesPlugin.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+		return teamCapesOverlay;
+	}
+
+	@Schedule(
+		period = 1800,
+		unit = ChronoUnit.MILLIS
+	)
+	public void update()
+	{
+		teams.clear();
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesPlugin.java]: Maybe move this part under the check? I know that clearing is not expensive on empty list, but it is still overhead without reason when the plugin is disabled.

File: runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesOverlay.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	public Dimension render(Graphics2D graphics, Point parent)
+	{
+		Map<Integer, Integer> teams = plugin.getTeams();
+		if (teams.isEmpty())
+		{
+			return null;
+		}
+		// sort teams by player count
+		List<Integer> keys = new ArrayList<>(teams.keySet());
+		keys.sort(Comparator.comparing(teams::get));
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesOverlay.java]: The sorting could be moved to update() so it only happens once.

**Comment by Adam-** [comment]
Thanks!

**Pull request merged**